program-runtime/src/invoke_context.rs

```diff
@@ -326,13 +326,14 @@ impl<'a> InvokeContext<'a> {
         instruction: &Instruction,
         signers: &[Pubkey],
     ) -> Result<(Vec<InstructionAccount>, Vec<IndexOfAccount>), InstructionError> {
-        // Finds the index of each account in the instruction by its pubkey.
-        // Then normalizes / unifies the privileges of duplicate accounts.
-        // Note: This is an O(n^2) algorithm,
-        // but performed on a very small slice and requires no heap allocations.
+        // We reference accounts by an u8 index, so we have a total of 256 accounts.
+        // This algorithm allocates the array on the stack for speed.
+        // On AArch64 in release mode, this function only consumes 640 bytes of stack.
+        let mut transaction_callee_map: [u8; 256] = [u8::MAX; 256];
+        let mut instruction_accounts: Vec<InstructionAccount> =
+            Vec::with_capacity(instruction.accounts.len());
         let instruction_context = self.transaction_context.get_current_instruction_context()?;
-        let mut deduplicated_instruction_accounts: Vec<InstructionAccount> = Vec::new();
-        let mut duplicate_indicies = Vec::with_capacity(instruction.accounts.len());
+
```

Comment:
We should also `debug_assert!(instruction.accounts.len() <= u8::MAX)` because we do `instruction_accounts.len() as u8` later.